File: task03/go/subtask4.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
)

func main() {
    file, err := os.Open("input.txt")
    if err != nil {
        fmt.Println(err)
        return
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    scanner.Scan()
    n, _ := strconv.Atoi(scanner.Text())

    file, err = os.Create("output.txt")
    if err != nil {
        fmt.Println(err)
        return
    }
    defer file.Close()

    writer := bufio.NewWriter(file)

    for i := 0; i < n; i++ {
        for j := 0; j < n - i - 1; j++ {

File: task03/js/subtask4.js
const fs = require('fs');

fs.readFile('input.txt', 'utf8', (err, data) => {
  if (err) {
    console.error(err);
    return;
  }
  const n = parseInt(data);

  let output = '';
  for (let i = 0; i < n; i++) {
    output += " ".repeat(n - i - 1) + "*".repeat(2 * i + 1) + "\n";
  }

  for (let i = n - 2; i >= 0; i--) {
    output += " ".repeat(n - i - 1) + "*".repeat(2 * i + 1) + "\n";
  }

  fs.writeFile('output.txt', output, (err) => {
    if (err) {
      console.error(err);
    }
  });
});